pkg/graph/serve.go

```diff
@@ -40,9 +40,11 @@ func NewGraph(logger *zap.Logger) graphInterface {
 const defaultPort = 6789
 
 // Serve is called by the serve command and is used to run a graphql server, to run tests separately via apis.
-func (g *graph) Serve(path string, proxyPort uint32, mongopassword, testReportPath string, Delay uint64, pid, port uint32, lang string, passThroughPorts []uint, apiTimeout uint64, appCmd string, enableTele bool) {
+func (g *graph) Serve(path string, proxyPort uint32, mongopassword, testReportPath string, generateTestReport bool, Delay uint64, pid, port uint32, lang string, passThroughPorts []uint, apiTimeout uint64, appCmd string, enableTele bool) {
 	var ps *proxy.ProxySet
 
+	defer pkg.DeleteTestReports(g.logger, generateTestReport)
```

Comment:
why exactly are we deleting the reports at the end of the function?

Reply:
serve command require testReports to show status. So I removed the testReports when function ends

Reply:
Can't we just get the status without writing to a file, or checking the condition before writing to the file?

Reply:
Actually this method was recommended here https://github.com/keploy/keploy/pull/1513#issuecomment-1931478001

Reply:
ok